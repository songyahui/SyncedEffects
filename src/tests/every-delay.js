"use hopscript"

var hh = require("hiphop");

var prg =
    <hh.module I=${{accessibility: hh.IN}} O>
      <hh.every countValue=2 I>
	<hh.emit O/>
      </hh.every>
    </hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "everydelay");
