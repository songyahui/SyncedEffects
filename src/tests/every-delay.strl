module everydelay:

input I;
output O;

every 2 I do
      emit O;
end every

end module
