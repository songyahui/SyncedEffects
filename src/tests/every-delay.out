everydelay> I;
--- Output:
everydelay> I;
--- Output:
everydelay> ;
--- Output:
everydelay> ;
--- Output:
everydelay> I;
--- Output: O
everydelay> ;
--- Output:
everydelay> I;
--- Output:
everydelay> I;
--- Output: O
everydelay> I;
--- Output:
everydelay> ;
--- Output:
